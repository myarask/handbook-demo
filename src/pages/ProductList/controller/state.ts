import { create } from "zustand";

export const useProductList = create<{
  searchTerm: string;
  pageSize: number;
  page: number;
}>()(() => ({
  searchTerm: "",
  pageSize: 10,
  page: 1,
}));
