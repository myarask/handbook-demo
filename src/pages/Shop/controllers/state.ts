import { create } from "zustand";

export const useShop = create<{
  searchTerm: string;
  pageSize: number;
  page: number;
  reviewing: {
    hasShownInterest: boolean;
    productId: number | null;
  };
  buying: {
    hasShownIntent: boolean;
    productId: number | null;
  };
}>()(() => ({
  searchTerm: "",
  pageSize: 10,
  page: 1,
  reviewing: {
    hasShownInterest: false,
    productId: null,
  },
  buying: {
    hasShownIntent: false,
    productId: null,
  },
}));
