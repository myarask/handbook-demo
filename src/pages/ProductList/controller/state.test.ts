import { useProductList } from "./state";

test("should initialize with default state", () => {
  const state = useProductList.getState();
  expect(state).toMatchSnapshot();
});
