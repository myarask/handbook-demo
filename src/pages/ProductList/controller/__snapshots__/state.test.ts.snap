// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should initialize with default state 1`] = `
{
  "page": 1,
  "pageSize": 10,
  "searchTerm": "",
}
`;
